extensions/keycloak-admin-client-common/runtime/src/main/java/io/quarkus/keycloak/admin/client/common/KeycloakAdminClientConfig.java

```diff
@@ -0,0 +1,72 @@
+package io.quarkus.keycloak.admin.client.common;
+
+import java.util.Optional;
+
+import io.quarkus.runtime.annotations.ConfigItem;
+import io.quarkus.runtime.annotations.ConfigPhase;
+import io.quarkus.runtime.annotations.ConfigRoot;
+
+/**
+ * Keycloak Admin Client
+ */
+@ConfigRoot(phase = ConfigPhase.RUN_TIME, name = "keycloak.admin-client")
+public class KeycloakAdminClientConfig {
+
+    /**
+     * Realm.
+     */
+    @ConfigItem(defaultValue = "master")
+    public String realm;
+
+    /**
+     * Keycloak server URL, for example, `https://host:port`.
+     */
+    @ConfigItem
+    public Optional<String> serverUrl;
+
+    /**
+     * Client id.
+     */
+    @ConfigItem
+    public Optional<String> clientId;
+
+    /**
+     * Username. Required with a `password` grant type.
+     */
+    @ConfigItem
+    public Optional<String> username;
+
+    /**
+     * Password. Required with a `password` grant type.
+     */
+    @ConfigItem
+    public Optional<String> password;
+
+    /**
+     * Client secret. Required with a `client_credentials` grant type.
+     */
+    @ConfigItem
+    public Optional<String> clientSecret;
+
+    /**
+     * OAuth 2.0 <a href="https://datatracker.ietf.org/doc/html/rfc6749#section-3.3">Access Token Scope</a>.
+     */
+    @ConfigItem
+    public Optional<String> scope;
+
+    /**
+     * OAuth Grant Type.
+     */
+    @ConfigItem
+    public Optional<GrantType> grantType;
```

Comment:
@michalvavrik I believe it has to be defaulted to `password` since the typical use is captured in this removed code:
```
    private Keycloak keycloak() {
        try {
            return KeycloakBuilder.builder()
                    .serverUrl(url)
                    .realm("master")
                    .clientId("admin-cli")
                    .username("admin")
                    .password("admin")
                    .build();
        } catch (Exception ex) {
            throw new RuntimeException(ex);
        }
```

where a `password` grant is used to get access to the `master` realm - from where other realms can be created.